pandora-client-web/src/components/settings/securitySettings.tsx

```diff
@@ -0,0 +1,273 @@
+					connections.map((c) => <ConnectedClientConnection key={ c.loginTokenId } connection={ c } />)
+				}
+			</tbody>
+		</table>
+	);
+}
+
+function ConnectedClientConnection({ connection }: { connection: AccountConnectedClients[number]; }): ReactElement {
+	const directoryConnector = useDirectoryConnector();
+	const authToken = useObservable(directoryConnector.authToken);
+	const confirm = useConfirmDialog();
+
+	const { loginTokenId, connectionCount, connectedCharacters } = connection;
+
+	const disconnect = React.useCallback(() => {
+		void confirm('Are you sure you want to disconnect this client?').then((confirmed) => {
+			if (confirmed) {
+				directoryConnector.sendMessage('logout', { type: 'selected', accountTokenId: loginTokenId });
+			}
+		});
+	}, [confirm, directoryConnector, loginTokenId]);
+
+	const isCurrent = authToken?.value.startsWith(loginTokenId);
+
+	return (
+		<tr className={ isCurrent ? 'current-connection' : undefined }>
+			<td>{ loginTokenId }</td>
+			<td>{ connectionCount }</td>
+			<td>{ connectedCharacters.map(({ id, name }) => `${name} (${id})`).join(', ') }</td>
+			<td>
+				{
+					isCurrent ? (
+						<>
+							<Button className='slim' onClick={ disconnect }>Logout</Button>
+							<ExtendCurrentSession token={ authToken! } />
+						</>
+					) : (
+						<Button className='slim' onClick={ disconnect }>Disconnect</Button>
+					)
+				}
+			</td>
+		</tr>
+	);
+}
+
+function ExtendCurrentSession({ token }: { token: AuthToken; }): ReactElement {
+	const [show, setShow] = React.useState(false);
+
+	const hide = React.useCallback(() => {
+		setShow(false);
+		return true;
+	}, []);
+
+	useKeyDownEvent(hide, 'Escape');
+
+	if (!show)
+		return <Button className='slim' onClick={ () => setShow(true) }>Extend</Button>;
+
+	return (
+		<>
+			<Button className='slim' onClick={ hide }>Extend</Button>
+			<ExtendCurrentSessionDialog token={ token } hide={ hide } />
+		</>
+	);
+}
+
+function ExtendCurrentSessionDialog({ token, hide }: { token: AuthToken; hide: () => boolean; }): ReactElement {
+	const directory = useDirectoryConnector();
+	const [password, setPassword] = React.useState('');
+	const now = useCurrentTime(60_000);
+
+	const [extend, processing] = useAsyncEvent(
+		() => directory.extendAuthToken(password),
+		(result) => {
+			if (result)
+				hide();
+			else
+				setPassword('');
+		},
+	);
+
+	return (
+		<ModalDialog>
+			<Form dirty={ false } onSubmit={ extend }>
+				<p>Your session will expire in { FormatTimeInterval(token.expires - now, 'short') }</p>
+				<FormField>
+					<label htmlFor='extend-current-session-password'>Password</label>
+					<input
+						type='password'
+						id='extend-current-session-password'
+						value={ password }
+						onChange={ (e) => setPassword(e.target.value) }
+					/>
+				</FormField>
+				<br />
+				<Row>
+					<Button onClick={ hide } disabled={ processing }>Cancel</Button>
+					<Button type='submit' disabled={ processing }>Extend</Button>
+				</Row>
```

Comment:
This dialog doesn't work at all. Was it even tested?
Clicking the submit button triggers a page reload, because the submit event does not prevent default.
As there is no code to handle to submit on URL, the page (almost always) reloads quicker than the handler for this runs (even sends to directory thanks to the need to hash things first), so the whole form does nothing...

Reply:
issue was that default form action was not prevented, should be good now